api/server/creds.go

```diff
@@ -11,61 +12,155 @@ import (
 		return
 	}
 
-	creds, err := d.CredsEnumerate()
+	// Get gRPC connection
+	conn, err := vd.getConn()
 	if err != nil {
 		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	json.NewEncoder(w).Encode(creds)
+
+	client := api.NewOpenStorageCredentialsClient(conn)
+	resp, err := client.Enumerate(ctx, &api.SdkCredentialEnumerateRequest{})
+
+	if err != nil {
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	credentials := make(map[string]interface{})
+
+	for _, credentialId := range resp.CredentialIds {
+		credResp, err := client.Inspect(r.Context(), &api.SdkCredentialInspectRequest{
+			CredentialId: credentialId,
+		})
+
+		if err != nil {
+			vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		credentials[credResp.CredentialId] = credResp
+	}
+
+	if err := json.NewEncoder(w).Encode(credentials); err != nil {
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
 }
 
 func (vd *volAPI) credsCreate(w http.ResponseWriter, r *http.Request) {
 	method := "credsCreate"
-	var input api.CredCreateRequest
-	response := &api.CredCreateResponse{}
+	var (
+		input api.CredCreateRequest
+		err   error
+	)
 
-	if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
+	if err = json.NewDecoder(r.Body).Decode(&input); err != nil {
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	useProxy := false
+
+	if len(input.InputParams[api.OptCredProxy]) != 0 {
+		useProxy, err = strconv.ParseBool(input.InputParams[api.OptCredProxy])
+
+		if err != nil {
+			vd.sendError(vd.name, method, w, err.Error(), http.StatusBadRequest)
+			return
+		}
+	}
+
+	ownerShip, err := getOwnershipFromString(input.InputParams[api.OptCredOwnership])
+
+	if err != nil {
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	sdkCreds, err := api.GetSdkCreds(input.InputParams)
+
+	if err != nil {
 		vd.sendError(vd.name, method, w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	d, err := vd.getVolDriver(r)
+
+	sdkReq := &api.SdkCredentialCreateRequest{
+		Name:           input.InputParams[api.OptName],
+		Bucket:         input.InputParams[api.OptCredBucket],
+		EncryptionKey:  input.InputParams[api.OptCredEncrKey],
+		UseProxy:       useProxy,
+		Ownership:      ownerShip,
+		CredentialType: sdkCreds,
+	}
+	ctx, err := vd.annotateContext(r)
+
 	if err != nil {
-		notFound(w, r)
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	response.UUID, err = d.CredsCreate(input.InputParams)
+	// Get gRPC connection
+	conn, err := vd.getConn()
 	if err != nil {
 		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	json.NewEncoder(w).Encode(response)
+
+	client := api.NewOpenStorageCredentialsClient(conn)
+	resp, err := client.Create(ctx, sdkReq)
```

Comment:
err is lost here. it is replaced by line 120. Please return if error is found

Reply:
gotcha